src/components/paster.ts

```diff
@@ -0,0 +1,201 @@
+
+        if (clipboardContents === '') {
+            return
+        }
+
+        if (clipboardContents.split('\n').length === 1) {
+            const fpath = path.resolve(fileUri.fsPath, clipboardContents)
+            if (fs.existsSync(fpath)) {
+                this.pasteFile(editor, fileUri.fsPath, clipboardContents)
+                return
+            }
+        }
+        // if not pasting file
+        try {
+            this.pasteTable(editor, clipboardContents)
+        } catch (error) {
+            this.pasteNormal(editor, this.reformatText(clipboardContents))
+        }
+    }
+
+    public pasteNormal(editor: vscode.TextEditor, content: string) {
+        editor.edit(edit => {
+            const current = editor.selection
+
+            if (current.isEmpty) {
+                edit.insert(current.start, content)
+            } else {
+                edit.replace(current, content)
+            }
+        })
+    }
+
+    public pasteFile(editor: vscode.TextEditor, baseFile: string, file: string) {
+        const TABLE_FORMATS = ['.csv']
+        const extension = path.extname(file)
+
+        if (TABLE_FORMATS.indexOf(extension) !== -1) {
+            if (extension === '.csv') {
+                const rows: string[] = []
+
+                fs.createReadStream(path.resolve(baseFile, file))
+                    .pipe(csv())
+                    .on('data', (data: Object) => rows.push(Object.values(data).join('\t')))
+                    .on('end', () => {
+                        const body = rows.join('\n')
+                        this.pasteTable(editor, body)
+                    })
+            }
+        }
+    }
+
+    public pasteTable(editor: vscode.TextEditor, content: string) {
+        this.extension.logger.addLogMessage('Pasting: Table')
+        // trim surrounding whitespace
+        content = content.replace(/^\s*/, '').replace(/\s*$/, '')
+        content = this.reformatText(content, false)
+        const lines = content.split('\n')
+        const cells = lines.map(l => l.split('\t'))
+
+        // determine if all rows have same number of cells
+        const isConsistent = cells.reduce((accumulator, current, _index, array) => {
+            if (current.length === array[0].length) {
+                return accumulator
+            } else {
+                return false
+            }
+        }, true)
+        if (!isConsistent) {
+            throw new Error('Table is not consistent')
+        } else if (cells.length === 1 && cells[0].length === 1) {
+            this.pasteNormal(editor, content)
+            return
+        }
+
+        const columnType: string = vscode.workspace.getConfiguration('latex-workshop.formattedPaste')['tableColumnType']
+        const booktabs: boolean = vscode.workspace.getConfiguration('latex-workshop.formattedPaste')['tableBooktabsStyle']
+        const headerRows: number = vscode.workspace.getConfiguration('latex-workshop.formattedPaste')['tableHeaderRows']
```

Comment:
You should call `vscode.workspace.getConfiguration('latex-workshop.formattedPaste')` once.

Reply:
Fixed in 4bbcc93175dd610259dfe12a4d131b1a5bfabfb6.